src/Feature/FeatureTrace.hs

```diff
@@ -47,6 +47,10 @@ simplifyFeatureTraceOfNodes f t d = \v -> if member v d then simplifyFeatureTrac
 Calculates the presence condition of a node (third argument) in the given tree (first argument) with the given feature traces (second argument).
 See Equation 1 in the paper.
 -}
+-- TODO(benjamin) In the paper `trace node` is not included if `node` is
+-- `Mandatory`. Does this code need the following invariant?
+-- `all (== Nothing) [trace n | n <- nodes root, optionaltype n == Mandatory]`
+-- If yes this has to be documented.
```

Comment:
I agree. Would be so kind and to it? I guess the right place to document this invariant would be at the definition of `FeatureTrace`.